lib/Doctrine/Common/DataFixtures/ReferenceRepository.php

```diff
@@ -94,27 +111,41 @@ protected function getIdentifier($reference, $uow)
      */
     public function setReference($name, $reference)
     {
+        $class = get_class($reference);
```

Comment:
If the object is a proxy object, this will register that reference under the name of the proxy class and not under the name of the entity, and it won't be found in `getReference()` when passing the entity class.

Reply:
I added a getRealClass call